packages/gatsby-cli/src/reporter/loggers/json/index.ts

```diff
@@ -0,0 +1,28 @@
+import { onLogAction } from "../../redux/index"
+import { ActionsUnion, ISetStatus } from "../../redux/types"
+import stripAnsi from "strip-ansi"
+import _ from "lodash"
+
+const isStringPayload = (action: ActionsUnion): action is ISetStatus =>
+  typeof action.payload === `string`
+
+const sanitizeAction = (action: ActionsUnion): ActionsUnion => {
+  if (isStringPayload(action)) {
+    return action
+  }
+
+  if (`text` in action.payload && action.payload.text) {
+    action.payload.text = stripAnsi(action.payload.text)
+  }
+  if (`statusText` in action.payload && action.payload.statusText) {
+    action.payload.statusText = stripAnsi(action.payload.statusText)
+  }
+
+  return action
+}
+
+onLogAction((action: ActionsUnion) => {
+  const sanitizedAction = sanitizeAction(action)
+
+  process.stdout.write(JSON.stringify(sanitizedAction) + `\n`)
+})
```

Comment:
We actually want to change this module to not invoke functionality on import. So this `onLogAction` should be wrapped in an exported function, which should then be executed in the parent module. Mind making these changes?

```
export function initializeJSONLogger() {
  onLogAction(....)
}
```

Reply:
I've changed this to named export `initializeJSONLogger`